packages/discord.js/src/structures/interfaces/TextBasedChannel.js

```diff
@@ -165,6 +165,14 @@ class TextBasedChannel {
       return dm.send(options);
     }
 
+    /**
+     * If `nonce` is not provided generate one & set `enforceNonce`, unless `enforceNonce` is explicitly set to `false`
+     */
+    if (options.nonce === undefined && options.enforceNonce !== false && this.client.options.enforceNonce) {
```

Comment:
This should be in `MessagePayload#resolveBody`

Reply:
Good catch! I had to make a couple of small logic changes to account for some of the existing code